interface/patient_file/encounter/find_code_history.php

```diff
@@ -0,0 +1,215 @@
+            let targetDoneButton = top.document.getElementById('saveDx');
+            let targetTipsButton = top.document.getElementById('showTips');
+            targetDoneButton.onclick = function () {
+                setFormDx();
+            };
+            targetTipsButton.onclick = function () {
+                $("#tips").toggle();
+            };
+        });
+    </script>
+</head>
+<body>
+    <div class="container-fluid">
+        <div id="tips" class="tips">
+            <section class="panel panel-default">
+                <header class="panel-heading panel-heading-sm">
+                    <h4 class="panel-title"><?php echo xlt('Usage Tips') ?></h4>
+                </header>
+                <div class="panel-body panel-body-sm">
+                    <ul>
+                        <?php
+                        echo "<li>" . xlt("This dialog is generated from patients problems diagnoses and the accumulated diagnoses of past procedures.") . "</li>";
+                        echo "<li>" . xlt("Build diagnoses list by clicking appropriate code button.") . "</li>";
+                        echo "<li>" . xlt("Duplicate codes are deleted from list, otherwise; appends to list.") . "</li>";
+                        echo "<li>" . xlt("Once finished editing, click Save. Procedure forms procedure diagnoses will fill exactly as built in this dialog.") . "</li>";
+                        echo "<li>" . xlt("The legacy code finder is still available for codes not found in this finder or codes list editing.") . "</li>";
+                        ?>
+                    </ul>
+                    <button class='btn btn-xs btn-success pull-right' onclick='$("#tips").toggle();return false;'><?php echo xlt('Dismiss') ?></button>
+                </div>
+            </section>
+        </div>
+        <div class="input-group">
+            <span class="input-group-addon" onclick='clearCodes(this)'><i class="fa fa-trash fa-1x"></i></span>
+            <input class='form-control' type='text' id='workingDx'
+                title='<?php echo xla('Current Working Procedure Diagnoses'); ?>' value='' />
+        </div>
+        <br>
+        <div>
+            <table class="table table-condensed table-striped">
+                <thead>
+                <tr>
+                    <th><?php echo xlt('Origin'); ?></th>
+                    <th><?php echo xlt('Code'); ?></th>
+                    <th><?php echo xlt('Description'); ?></th>
+                </tr>
+                </thead>
+                <tbody>
+                <?php
+                foreach ($dxcodes as $pc) {
+                    echo "<tr>\n" .
+                        "<td>" . $pc['origin'] . "</td>\n" .
+                        "<td><button class='btn btn-xs btn-default' onclick='rtnCode(this)' value='" . attr($pc['code']) . "'>" .
+                        text(explode(':', $pc['code'])[1]) . "</button></td>\n" .
+                        "<td>" . $pc['desc'] . "</td>\n" .
```

Comment:
yes, got one :)
This should be escaped with text()

Reply:
lol, Hey it wouldn't be a PR from Jerry unless missing an escape!